### PR TITLE
If no auth redirect to login

### LINKED ISSUE #79: Redirect to login page when unauthorized
Currently it lets you see everything, but fails on submission.

### PR DESCRIPTION
**Pull request by aguspina**
Fixes #79

### CODE REVIEW

File: app/src/App.js [hunk 1]
```diff
@@ -96,16 +99,22 @@ angular
...
-      var token = store.get('token');
-      if (token) {
-        if (!jwtHelper.isTokenExpired(token)) {
-          if (!auth.isAuthenticated) {
-            auth.authenticate(store.get('profile'), token);
+  .run(function($rootScope, $state, $timeout, auth, store, jwtHelper, navigationService) {
+    $rootScope.$on('$stateChangeStart', function(ev, toState) {
+      if(toState.needsAuthentication) {
+        var token = store.get('token');
+        if (token) {
```
Comment (faloi) [inline app/src/App.js]: This is a copy-paste from auth0 docs, isn't it? I know you can do better than this, "Franco".

Comment (aguspina) [inline app/src/App.js]: The jwt token control was not added in this commit. Just the needsAuthenthication property, and the prevent default.

Comment (aguspina) [inline app/src/App.js]: By the way, I did this. But for a weird reason, I've franco's gitconfig :dizzy_face:

Comment (faloi) [inline app/src/App.js]: Ok, but this should be encapsulated in a near future, see how I've done it [here](https://github.com/pepita-remembrance/Grupo-A-022015-client/commit/c5d5889910e666d0f3de4c0af1c7dd9016ed9586) :stuck_out_tongue:

**Pull request merged**